service/src/main/python/jcarbon/tensorflow/callbacks.py

```diff
@@ -125,6 +97,98 @@ def on_epoch_end(self, epoch, logs=None):
         self.reports[epoch] = self.last_report.to_frame().assign(
             epoch=epoch).set_index('epoch', append=True)
 
+class JCarbonExperimentCallback(JCarbonChunkingCallback):
+    def __init__(
+            self,
+            addr='localhost:8980',
+            period_ms=DEFAULT_PERIOD_MS,
+            signals=DEFAULT_SIGNALS,
+            chunking_period_sec=DEFAULT_PERIOD_SECS):
+        super().__init__(addr, period_ms, signals)
+        self.timestamps = {}
+        self.batch_timestamps = None
+
+    def on_epoch_begin(self, epoch, logs=None):
+        super().on_epoch_begin(epoch, logs)
+        self.curr_batch_timestamps = None
+    
+    def on_train_batch_begin(self, batch, logs=None):
+        self.batch_start = time.time()
+        super().on_train_batch_begin(batch, logs)
+        
+        
+    def on_train_batch_end(self, batch, logs=None):
+        curr = time.time()
+        super().on_train_batch_end(batch, logs)
+        self.batch_end = int((10**9 * curr))
+        self.batch_start = int((10**9 * self.batch_start))
+        if self.curr_batch_timestamps is None:
+            self.curr_batch_timestamps = [{'batch': batch, 'start': self.batch_start, 'end': self.batch_end}]
+        else:
+            self.curr_batch_timestamps.append({'batch': batch, 'start': self.batch_start, 'end': self.batch_end})
```

Comment:
This entire block can be replaced with appending to `self.batch_timestamps` based on my above comments.